<div style="text-align: center;">
  <img src="https://github.com/Hack-io-Data/Imagenes/blob/main/01-LogosHackio/logo_naranja@4x.png?raw=true" alt="esquema" />
</div>

# Laboratorio Agrupamiento de Datos

En este laboratorio usaremos el DataFrame creado en la lección anterior resultado de la unión de los conjuntos de datos compartidos contigo. 

**Instrucciones:**

1. Lee cuidadosamente el enunciado de cada ejercicio.

2. Implementa la solución en la celda de código proporcionada.

3. Devuelve los resultados siempre en formato de DataFrame

In [1]:
# Carga el dataset de trabajo
import pandas as pd
import numpy as np

df_Netflix = pd.read_csv ("datos/netflix_merged.csv")
df_Netflix.head()

Unnamed: 0            Title                  Genre           Premiere  \
0           0  Enter the Anime            Documentary     August 5, 2019   
1           1      Dark Forces               Thriller    August 21, 2020   
2           2          The App  Science fiction/Drama  December 26, 2019   
3           3   The Open House        Horror thriller   January 19, 2018   
4           4      Kaali Khuhi                Mystery   October 30, 2020   

   Runtime  IMDB Score          Language show_id   type            title  \
0       58         2.5  English/Japanese   s3624  Movie  Enter the Anime   
1       81         2.6           Spanish   s2105  Movie      Dark Forces   
2       79         2.6           Italian   s3095  Movie          The App   
3       94         3.2           English   s5074  Movie   The Open House   
4       90         3.4             Hindi   s1778  Movie      Kaali Khuhi   

                    director  \
0              Alex Burunova   
1          Bernardo Arellano   
2               Elisa Fuksas   
3  Matt Angel, Suzanne Coote   
4            Terrie Samundra   

                                                cast                country  \
0  Kozo Morishita, Yoko Takahashi, Shinji Aramaki...   United States, Japan   
1  Tenoch Huerta, Eréndira Ibarra, Dale Carley, A...                 Mexico   
2  Vincenzo Crea, Jessica Cressy, Greta Scarano, ...                  Italy   
3  Dylan Minnette, Piercey Dalton, Patricia Bethu...  Canada, United States   
4  Shabana Azmi, Leela Samson, Sanjeeda Sheikh, S...                  India   

          date_added  release_year rating duration  \
0     August 5, 2019          2019  TV-MA   59 min   
1    August 21, 2020          2020  TV-MA   81 min   
2  December 26, 2019          2019  TV-MA   79 min   
3   January 19, 2018          2018  TV-MA      NaN   
4   October 30, 2020          2020  TV-14      NaN   

                                           listed_in  \
0                Documentaries, International Movies   
1  Horror Movies, Independent Movies, Internation...   
2   Dramas, Independent Movies, International Movies   
3                           Horror Movies, Thrillers   
4                Horror Movies, International Movies   

                                         description  
0  What is anime? Through deep-dives with notable...  
1  In search of his sister, a renegade criminal s...  
2  While in Rome to shoot his first movie, actor ...  
3  Following a tragedy, a mother and her teen son...  
4  When a restless spirit curses a village that h...

#### Ejercicio 1: ¿Cuántos títulos hay por cada tipo de contenido ('Movie' o 'TV Show')?

In [13]:
df_ej = df_Netflix.groupby('type').size().reset_index(name='count')
df_ej


type  count
0  Movie    513

#### Ejercicio 2: ¿Cuál es el promedio de puntuación de las películas agrupadas por clasificación de audiencia ('rating')? Interpreta los resultados, ¿hay algo que te llame la atención? Explica por qué. 

In [10]:
condicion = df_Netflix['type'] == 'Movie'
df_calificacion = df_Netflix[condicion].groupby('rating')['IMDB Score']
df_promedio = df_calificacion.mean().reset_index()
df_promedio


rating  IMDB Score
0      G    6.500000
1     PG    6.800000
2  PG-13    6.386957
3      R    6.495745
4  TV-14    6.177660
5   TV-G    6.156250
6  TV-MA    6.097638
7  TV-PG    6.308772
8   TV-Y    6.780000
9  TV-Y7    6.240000

Observamos que al agrupar las películas por clasificación de audiencia y rating se muestra que las películas con rating PG tienen una media mayor que 6.8, mientras que las de rating TV-14 tienen el promedio más bajo con 6.18. Tal vez esto indique que las peliculas PG dirigidas a un público más amplio y familiar tiedan a recibir mejor calificaciónque las de rating más restrictivo

#### Ejercicio 3: ¿Cuántos títulos hay por cada país? Ordena los resultados de mayor a menor

In [25]:
df_por_pais=df_Netflix.groupby('country')
df_conteo_por_pais=df_por_pais.size().reset_index(name='count')
df_conteo_por_pais_ordenado = df_conteo_por_pais.sort_values(by='count', ascending=False)
df_conteo_por_pais_ordenado


country  count
67                           United States    274
22                                   India     35
59                          United Kingdom     13
30                                   Italy     13
50                                   Spain     12
..                                     ...    ...
73                  United States, Germany      1
75                    United States, Japan      1
78   United States, United Kingdom, Canada      1
79  United States, United Kingdom, Germany      1
80    United States, United Kingdom, India      1

[81 rows x 2 columns]

#### Ejercicio 4: ¿Cuál es la calificación promedio en IMDb de los títulos agrupados por tipo ('Movie' o 'TV Show')?

#### Ejercicio 5: ¿Cuántos títulos se añadieron a Netflix cada año? Para resolver este ejercicio tendrás uqe convertir la columna `date_added` a formato fecha usando el método `pd.to_datetime`. [Aquí](https://pandas.pydata.org/docs/reference/api/pandas.to_datetime.html) tienes la documentación, leeala con detenimiento antes de usar el método. 

#### Ejercicio 6: ¿Cuál es el país con más títulos lanzados en el año 2020? Para resolver este ejercicio después del gropuby os puede resultar interesante el método [`idxmax`](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.idxmax.html)

#### Ejercicio 7:¿Cuál es el promedio de puntuación en IMDb de los títulos agrupados por país?

#### Ejercicio 8: ¿Cuántos títulos fueron lanzados en cada año agrupados por tipo ('Movie' o 'TV Show')?

#### Ejercicio 9: ¿Cuál es el país con la mayor cantidad de títulos de clasificación 'TV-MA'?

#### Ejercicio 10. ¿Cuál es el promedio de duración de las películas agrupadas por país?

#### Ejercicio 11: ¿Cuál es el número total de títulos por país y clasificación de audiencia ('rating')?

#### Ejercicio 12: ¿Cuántos títulos fueron lanzados antes del año 2000 agrupados por país?

#### Ejercicio 13: ¿Cuál es la puntuación promedio en IMDb de los títulos lanzados en 2021 agrupados por país?

#### Ejercicio 14: ¿Cuántos títulos fueron dirigidos por cada director? Ordena los resultados de mayor a menor

#### Ejercicio 15: ¿Cuál es el lenguaje más frecuente en los títulos lanzados en los últimos 5 años agrupado por país?